src/components/TextInput.tsx

```diff
@@ -0,0 +1,160 @@
+import { IconType } from "react-icons";
+import {
+    BiLockAlt,
+    BiAt,
+    BiCheese,
+    BiCreditCard,
+    BiDollar,
+    BiEuro,
+    BiLogoReddit,
+    BiSolidMessageAltError,
+    BiCheck,
+} from "react-icons/bi";
+
+type IconOption = "lock" | "at" | "cheese" | "card" | "dollar" | "euro" | "reddit" | "error" | "check";
+
+
+const iconLookup = (name: string | undefined): IconType | null => {
```

Comment:
instead of using a function, just use a regular hash map (e.g. JS object)

const icons = {
        lock: BiLockAlt,
        at: BiAt,
        cheese: BiCheese,
        card: BiCreditCard,
        dollar: BiDollar,
        euro: BiEuro,
        reddit: BiLogoReddit,
        error: BiSolidMessageAltError,
        check: BiCheck,
    };

Reply:
then access is as simple as `icons["lock"]` or

```
const prefixIcon = "lock"
icons[prefixIcon]
```